### PR TITLE
feat: add fomo list to the desktop version of the app

### LINKED ISSUE #10549: Add fomo list to desktop trade page
We should add the same fomo list that we have on mobile to desktop. <br><br>[Mockup here](https://www.figma.com/design/QWwFEart2z6DiGFdGUlti9/Site-Structure?node-id=2769-1766&t=TN5nulxn7l2aiSAe-0)<br><br>AC:

* List works just like mobile with a list that shows cards of assets from different lists 
* The list should be a carousel that moves by one card every 60 seconds
* On initial load we should animate in the list, and then stagger load them in move up and fade in
* Clicking a card loads that asset into the swapper the same behavior as mobile
* User can change the list by clicking on the name of the list in a popover
* User can change the sort order by clicking the filter icon this shows the popover

### PR DESCRIPTION
**Pull request by reallybeard**
## Description

Added a "fomo list" carousel at the bottom of the desktop trade page that shows popular/top assets. The carousel displays assets with their current price and 24h price change, and includes filtering options for different categories (Top Movers, OneClickDefi, Market Cap, Recently Added). Users can also sort and order the assets, and clicking on any asset will set it as the buy asset in the trade form.

The carousel only appears on desktop (hidden on mobile) and automatically hides once the user starts entering an amount in the trade form. It uses the same data sources and filtering logic as the markets page, so the categories and sorting options are consistent across the app.

## Issue (if applicable)

closes #10549 

## Risk

> High Risk PRs Require 2 approvals

This is a UI-only change that adds a new carousel component to the trade page. It doesn't modify any transaction logic, wallet interactions, or contract calls. The component reads from existing market data queries and only updates the trade input state when an asset is clicked (which is the same behavior as the existing asset search).

The changes are isolated to:
- New carousel component and card component
- TradeTab layout adjustments for spacing
- No changes to any swapper logic, transaction execution, or wallet interactions

**Risk Level: Low** - This is a view-only feature that doesn't touch any critical paths.

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

None. This is purely a UI enhancement that displays market data and allows users to select assets. All the underlying trade execution, wallet connections, and contract interactions remain unchanged.

### Engineering

1. **Desktop Testing:**
   - Navigate to the trade page on desktop (window width > md breakpoint)
   - Verify the carousel appears at the bottom of the page
   - Check that assets are displayed with icons, symbols, prices, and 24h change indicators
   - Verify the carousel auto-scrolls smoothly

2. **Filtering & Sorting:**
   - Click the settings icon on the right side of the carousel
   - Test changing categories (Top Movers, OneClickDefi, Market Cap, Recently Added)
   - Test sorting options (Price Change, Market Cap, APY, etc.)
   - Test order direction (Ascending/Descending)
   - Verify the carousel updates with the new filters

3. **Asset Selection:**
   - Click on any asset in the carousel
   - Verify it sets that asset as the buy asset in the trade form
   - Check Mixpanel events are firing correctly

4. **Responsive Behavior:**
   - Resize window to mobile size (< md breakpoint)
   - Verify carousel is hidden on mobile
   - Enter an amount in the trade form
   - Verify carousel hides when amount is entered
   - Clear the amount and verify carousel reappears

5. **Edge Cases:**
   - Test with slow network (verify loading states)
   - Test with no market data available
   - Test category switching when data is loading

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

**Testing Steps:**

1. **Visual Verification:**
   - Open the trade page on desktop
   - Confirm the carousel appears at the bottom with a list of assets
   - Verify the carousel scrolls automatically
   - Check that asset cards show icons, symbols, prices, and green/red arrows for price changes

2. **Functionality:**
   - Click the settings icon on the right side of the carousel
   - Try changing the category dropdown (should see different assets appear)
   - Try changing the sort by dropdown
   - Try changing the order (ascending/descending)
   - Click on any asset card - it should populate that asset in the "Buy" field of the trade form

3. **Responsive:**
   - Resize browser to mobile width - carousel should disappear
   - On desktop, type an amount in the trade form - carousel should disappear
   - Clear the amount - carousel should reappear

4. **Regression:**
   - Verify existing trade functionality still works (selecting assets, entering amounts, executing trades)
   - Verify the markets page still works correctly (since we're reusing the same data hooks)

## Screenshots (if applicable)
https://jam.dev/c/4dad90f5-9ec0-40bb-92e9-9a5c5f4b59cb

## Summary by CodeRabbit

* **New Features**
  * Top assets carousel with auto-scroll, animated slides, responsive display, filters (category/sort/order), and a new top-asset card for quick selection.
  * New hooks and a settings toggle to show/hide the Top Assets Carousel (persisted preference).

* **Bug Fixes**
  * Improved carousel click/drag behavior to prevent unintended interactions while dragging.

* **Style**
  * Updated hover color for raised surfaces in dark mode.

* **Chores**
  * Upgraded carousel library to latest major version and added related translations.

### CODE REVIEW

**Comment by premiumjibles** [comment]
@reallybeard  Wow this is insane :laughing: :fire: . And the animations are very impressive. Functionally works well! I'll dive into some code feedback next (if there is any) but first just some producty thoughts. None of this is blocking as you make the final call here but just some thoughts. 

1. The visual hierarchy here feels a little bit confusing with the icons and spacing. We are spacing more between the elements than within it which is good. But it still feels a bit disjointed and not super clear which element is which. One issue might be the double icon which makes each item less visually distinct. I'd be inclined to drop the arrow icon and instead just have have the price change text. maybe in brackets. I'm thinking the green vs. red with a - should be clear enough it's a drop or increase? Also maybe a teensy bit more space between items.

image

2. In general this feature is probably going to annoy people who don't like attention grabby things. This looks a bit like a breaking news bulletin which is part of it's charm but also I think a portion of users would appreciate the ability to hide it (which we can store in the preferencesSlice). It should be pretty easy to put a show/hide option in here. Perhaps we could also slow it down a bit to make it less hectic.

image

3. You can change order by, sort by, etc... but that's not super helpful when you can't actually see where the list begins and end. I'm not sure exactly what but it would be cool if there was a visual indicator that shows where the list ends/starts

4. The configuration popover UI in general feels a little bit unpolished. I'm guessing that's intentional because it's not very important but perhaps some little touches would be nice like chevrons to show it's a popover rather than a button. 

image

5. It feels a bit jarring that when you change the category the entire carousel including config button vanishes for a bit until the new list is ready. It would be nice if the carousel stayed but showed a loader

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetCard.tsx [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+  )
+  const changePercent24Hr = marketData?.changePercent24Hr
+
+  const isPositive = bnOrZero(changePercent24Hr).gt(0)
+
+  const handleClick = useCallback(() => {
+    onClick(asset)
+  }, [asset, onClick])
+
+  const handleKeyDown = useCallback(
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetCard.tsx]: Nitpick: Is this for accessibility? I don't think our app is very accessible to warrant special care for that. I'd be inclined to strip it out in the name of simplicity if that's the primary reason.

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetCard.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+      }
+    },
+    [asset, onClick],
+  )
+
+  return (
+    <Flex
+      cursor='pointer'
+      px={2}
+      height='44px'
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetCard.tsx]: Nitpick: Does this need to be in px or should it be defined in regular grid units (11)

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 3]
```diff
@@ -0,0 +1,562 @@
...
+  category: MarketsCategories
+  selectedCategory: MarketsCategories
+  onCategoryChange: (category: MarketsCategories) => void
+  checkedIcon: React.ReactElement
+}
+
+const CategoryMenuItem = ({
+  category,
+  selectedCategory,
+  onCategoryChange,
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: nitpick: if the components only job with onCategoryChange is to call onCategoryChange(category) it's probably 1% cleaner and easier to reason about if it's got an `onClick` callback and the parent deals with onCategoryChange(category)

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 4]
```diff
@@ -0,0 +1,562 @@
...
+  checkedIcon,
+}: CategoryMenuItemProps) => {
+  const translate = useTranslate()
+  const handleClick = useCallback(() => {
+    onCategoryChange(category)
+  }, [category, onCategoryChange])
+
+  const label = useMemo(() => {
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: nitpick: little bit hacky but not a big deal. If we could avoid the below special case i'd be inclined to delete this special sub component and just inline the MenuItemOption for a simpler mental model. But if it's crucial OneClickDefi have a special filter name that doesn't align with it's regular title then this is ok

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 5]
```diff
@@ -0,0 +1,562 @@
...
+  const categoryHook = useMemo(
+    () =>
+      selectedCategory === MarketsCategories.OneClickDefi
+        ? CATEGORY_TO_QUERY_HOOK[MarketsCategories.Trending]
+        : CATEGORY_TO_QUERY_HOOK[selectedCategory],
+    [selectedCategory],
+  )
+
+  const { data: categoryQueryData, isLoading: isCategoryQueryDataLoading } = categoryHook({
+    enabled: selectedCategory !== MarketsCategories.OneClickDefi,
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: q: could we ditch the special snowflake that is one click defi in this particular feature to simplify the code?

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 6]
```diff
@@ -0,0 +1,562 @@
...
+  const oneClickDefiAssets = useMemo(() => {
+    if (selectedCategory !== MarketsCategories.OneClickDefi || !portalsAssets) return []
+
+    return portalsAssets.ids
+      .slice(0, TOP_ASSETS_LIMIT)
+      .map(id => assetsById[id])
+      .filter(isSome)
+  }, [selectedCategory, portalsAssets, assetsById])
+
+  const categoryAssets = useMemo(() => {
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: non-blocking: Trying to think of ways to make this component less of a behemoth. Perhaps breaking out this asset fetching and parsing logic into a custom hook

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 7]
```diff
@@ -0,0 +1,562 @@
...
+  ])
+
+  const handleCategoryChange = useCallback(
+    (category: MarketsCategories) => {
+      dispatch(preferences.actions.setHighlightedTokensSelectedCategory(category))
+
+      switch (category) {
+        case MarketsCategories.TopMovers:
+          dispatch(
+            preferences.actions.setHighlightedTokensSelectedSort(SortOptionsKeys.PriceChange),
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: nitpick: Better if we had a map from MarketsCategories -> SortOptionsKeys and just had the one dispatch. Or even better if we just didn't have a SortOptionsKeys at all and just used the MarketsCategories as the sort selection

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 8]
```diff
@@ -0,0 +1,562 @@
...
+          paddingLeft: '0.5rem',
+        }}
+        variants={cardVariants}
+      >
+        <TopAssetCard asset={asset} onClick={handleRowClick} />
+      </motion.div>
+    ))
+  }, [popularAssets, handleRowClick])
+
+  const categoryLabel = useMemo(() => {
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: Nitpick: duplicated logic with the category item stuff, ideally consolidate but not crucial

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 9]
```diff
@@ -0,0 +1,562 @@
...
+      loop: true,
+      dragFree: true,
+      align: 'start',
+      slidesToScroll: 1,
+    },
+    [autoScrollPlugin],
+  )
+
+  useEffect(() => {
+    setIsVisible(true)
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: preferably blocking: what's the intention behind this? Wouldn't we just set the initial useState as useState(true)?

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 10]
```diff
@@ -0,0 +1,562 @@
...
+  }, [])
+
+  useEffect(() => {
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: Preferably-blocking: Seems like these isVisible useEffects are just a way of doing

`const isVisible = popularAssets.length !== 0` if i'm not mistaken? setIsVisible is only being called in response to data changes (not user driven) so i don't think it actually needs to be a state variable.

File: src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx [hunk 11]
```diff
@@ -0,0 +1,562 @@
...
+              variants={containerVariants}
+              initial='hidden'
+              animate={isVisible && popularAssets.length > 0 ? 'visible' : 'hidden'}
+            >
+              {popularAssetsCards}
+            </motion.div>
+          </Box>
+        </Box>
+        <Flex
+          height='44px'
```
Comment (premiumjibles) [inline src/components/MultiHopTrade/components/TradeInput/components/TopAssetsCarousel.tsx]: Preferably-blocking: Ideally don't hardcode the height of both the parent and the individual cards (both currently set to 44px). Either the cards should be hardcoded and the parent follows with a flex wrap OR the other way around.

**Comment by reallybeard** [comment]
@premiumjibles I've moved the fetching into its own hook, and the filter logic into its own hook. Then turned the popover menu into its own component. This reduced that component down to just 200 lines so its a lot more manageable now. 

Also should make it easy to plug into the HighlightedTokens component which this was a straight carbon copy of. But that would be better as its own PR I think.

Not sure what to do with the popover filters/sort as they are pretty basic.

**Review by premiumjibles — APPROVED** [review]
Happy with that 👌 , items looks more distinct and loader looks a lot less jarring. Up to you whether you want to add show/hide and start of list indication
